packages/angular/src/components/search/search-input.ts

```diff
@@ -1,18 +1,33 @@
-import { Component } from '@angular/core'
+import { afterNextRender, Directive, ElementRef, inject } from '@angular/core'
+import { logIfDevMode } from '../../utils/log-if-devmode'
 import { Input } from '../input'
 
 /**
  * Search input
  *
  * Used within SearchComponent to provide a search input.
  */
-@Component({
-  selector: 'input[ksd-search-input]',
+@Directive({
+  selector: 'input[ksdSearchInput]',
   standalone: true,
-  template: ` <ng-content /> `,
   host: {
     type: 'search',
     placeholder: '', // Need empty placeholder to enable show/hide for clear button
   },
 })
-export class SearchInput extends Input {}
+export class SearchInput {
+  private readonly input = inject(ElementRef<Input>)
+
+  constructor() {
```

Comment:
We are not really doing anything useful here, so we could maybe just document ksd-input + type=search, but it could be useful to encapsulate search-input so I think we just keep it. 

Can we just add the `ds-input` class also here then and remove this check?